### PR TITLE
Make sure feature telemetry state updates can be done in quick succession

### PR DESCRIPTION
**Pull request by IhorHandziuk**
**Problem:**
If feature telemetry state update calls are done quicker than it takes to start a session (a few milliseconds), then the update might not go through correctly.

**Solution:**
If the same feature telemetry instance needs to update the state of a feature more than once in a row, then make sure that the previous session had time to complete.

### CODE REVIEW

File: src/System Application/App/Telemetry/src/Feature Telemetry/FeatureUptakeStatusImpl.Codeunit.al [hunk 1]
```diff
@@ -139,4 +144,24 @@ codeunit 8705 "Feature Uptake Status Impl."
...
+    local procedure WaitForStartedUpdateFeatureUptakeSession()
+    var
+        StartDateTime: DateTime;
+        Timeout: Duration;
+    begin
+        if StartedSessionId = 0 then
+            exit;
+
+        Timeout := 100; // wait for up 0.1 seconds
+        StartDateTime := CurrentDateTime;
```
Comment (stkillen) [inline src/System Application/App/Telemetry/src/Feature Telemetry/FeatureUptakeStatusImpl.Codeunit.al]: Minor: Parentheses missing

Comment (IhorHandziuk) [reply src/System Application/App/Telemetry/src/Feature Telemetry/FeatureUptakeStatusImpl.Codeunit.al]: Added

File: src/System Application/App/Telemetry/src/Feature Telemetry/FeatureUptakeStatusImpl.Codeunit.al [hunk 2]
```diff
@@ -13,6 +13,11 @@ codeunit 8705 "Feature Uptake Status Impl."
     TableNo = "Feature Uptake";
     Permissions = tabledata "Feature Uptake" = rimd;
 
+    var
+        StartedSessionId: Integer;
```
Comment (AndreasMoth) [inline src/System Application/App/Telemetry/src/Feature Telemetry/FeatureUptakeStatusImpl.Codeunit.al:17]: Would it make sense to make this codeunit single instance instead? That way the benefit is applied to the entire session.

Comment (IhorHandziuk) [reply src/System Application/App/Telemetry/src/Feature Telemetry/FeatureUptakeStatusImpl.Codeunit.al:17]: If we were to go down that path, we would probably want to add some additional checks to make sure we are not waiting for a session that updates an unrelated feature.
Overall, I don't think this bug fix is worth that big of an investment.

**Pull request merged**